designs/sam_validate.md

```diff
@@ -0,0 +1,58 @@
+# `sam validate` command
+
+This is the design to improve the `sam validate` command to utilise the the `cfn-python-lint` package.
+
+It would be good to go beyond just implementing the CloudFormation package for validation of template files, there is the opportunity to create something better.
+
+## What's the current problem?
+
+The current iteration of `sam validate` only goes so far in validating the the SAM template, it does some cursorary checks:
+
+- Checks `CodeUri` to ensure an S3 URL exists
+- Checks if the template is valid but doesn't return errors
+  - If the template contains other resources outside of SAM it won't validate them
+
+## What will be changed?
+
+We will change how the `sam validate` command works, by integrating the [cfn-lint python package](https://github.com/awslabs/cfn-python-lint/) into the the validation workflow. This change would enhance the user's experience of working with SAM templates.
+
+## Success criteria for the change
+
+The user will be able to find detailed information about their template's validation errors or warnings by running `sam validate`. The validate command will also find any issues in non-SAM resources, such as S3 buckets or SNS topics.
+
+## Out of Scope
+
+- Customisation of the rules through the use of flags
```

Comment:
In the implementation section, it would be good to call out any default flags we will configure. We will probably want documentation for what they are as well.